enrichments/trace/internal/elastic/span.go

```diff
@@ -417,15 +422,38 @@ func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
 }
 
 type spanEventEnrichmentContext struct {
-	exception bool
+	exception        bool
+	exceptionEscaped bool
+
+	exceptionType       string
+	exceptionMessage    string
+	exceptionStacktrace string
 }
 
 func (s *spanEventEnrichmentContext) enrich(
 	se ptrace.SpanEvent,
 	cfg config.SpanEventConfig,
 ) {
+	// TODO(lahsivjar): Is error/log context handling be done in es-exporter?
+	// Ref: https://github.com/elastic/apm-data/blob/59d33b8113d629f43dc82f9b3931ceda63a19a7a/input/otlp/traces.go#L1243-L1260
```

Comment:
[**For reviewers**] APM-data sets context for span-events where it sets fields like `span.id`, `transaction.id`, `parent.id` based on the span ([ref](https://github.com/elastic/apm-data/blob/main/input/otlp/traces.go#L1243-L1260)), however, I am not sure if enrichments is the correct place to do this. Since the es-exporter could decide how to encode span-events into ES docs (separate or in the same doc as parent span) maybe it would be better to have this handled in the expoter. I am confused here, WDYT?

Reply:
Yes, I think that all properties that are already mapped as top level fields in the [logs data stream](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/logs-otel%40mappings.yaml) should not be handled by enrichments but in the ES exporter.

Reply:
CC: @carsonip This would require some changes in the spanevent handling in es-exporter. I think just copying some attributes from the parent span should be good enough.

Reply:
makes sense

Reply:
span event span.id, transaction.id are already done in elasticsearchexporter

Reply:
@carsonip Not sure if I am reading something incorrectly but I don't see all the fields handled in the exporter: https://github.com/elastic/apm-data/blob/59d33b8113d629f43dc82f9b3931ceda63a19a7a/input/otlp/traces.go#L1243-L1273, for example transaction type.

Reply:
transaction.type is not defined in mapping spec: https://github.com/elastic/opentelemetry-dev/blob/main/docs/design-decisions/ingest/mapping.md , that's why es exporter did not have specific handling for them. Do we actually need transaction.type for span events? If so, should they be in attributes.* or at top level?

Reply:
> Do we actually need transaction.type for span events?

I don't know

> If so, should they be in attributes.* or at top level?

If needed, it should be under attributes.

Reply:
> Yes, I think that all properties that are already mapped as top level fields in the [logs data stream](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/logs-otel%40mappings.yaml) should not be handled by enrichments but in the ES exporter.

> If needed, it should be under attributes.

As they are under attributes, it is better done in enrichment

Reply:
> As they are under attributes, it is better done in enrichment

I don't think it is a good idea to do this as enrichments since the fields we are talking about here are derived from the span. The ES exporter decides HOW to extract and encode the span event (in a separate doc or in the same doc). Whlie currently we are encoding them as a separate doc making this dependent on enrichment doesn't feel correct to me.

(https://github.com/elastic/opentelemetry-lib/pull/99#discussion_r1772810837 discussion thread is also related to the same topic)

Reply:
Me and @carsonip talked about this IRL and decided to do it as enrichments for now due to the differences in the handling of error and log contexts in apm-data ([ref](https://github.com/elastic/apm-data/blob/bd5e34116889942c53b186deab0f76beaf182de0/input/otlp/traces.go#L1243-L1273)). Once the specs are a bit more clearer around this we can revisit the enrichment.

Reply:
> https://github.com/elastic/opentelemetry-lib/pull/99/commits/222386e688fef526a691c98c0eb0569569c9e255

The above commit adds `transaction.{type, sampled}` enrichment for span event.